.github/workflows/tests-thorough.yml

```diff
@@ -15,16 +15,15 @@ jobs:
       fail-fast: false
     runs-on: ${{ matrix.os }}
     steps:
-      - uses: actions/checkout@v3
-      - uses: actions/setup-java@v3
+      - uses: actions/checkout@v4
+      - uses: actions/setup-java@v4
         with:
           distribution: 'zulu'
           java-version: 17
-          cache: 'maven'
-      - uses: gradle/gradle-build-action@v2
+      - uses: gradle/actions/setup-gradle@v3
         with:
           gradle-home-cache-cleanup: true
-          arguments: clean
+      - run: ./gradlew clean
```

Comment:
I think this `gradle clean` step can be removed. The project is freshly checked out every time, so there's nothing to clean..